deisctl should more gracefully handle spaces in config set

Updated stateless-platform 1.9.1 => 1.11.1

```
kalbasit@cratos ~ {dailymotion}  % deisctl config logs set storageAdapterType memory
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/deis/deis/deisctl/config.doConfigSet(0xde72d8, 0xc820262020, 0xc8200f3988, 0xb, 0xc82043e540, 0x2, 0x2, 0x0, 0x0, 0x0, ...)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/deisctl/config/config.go:73 +0x46f
github.com/deis/deis/deisctl/config.doConfig(0x7fff5fbff54f, 0x4, 0x75bce8, 0x3, 0xc82043e540, 0x2, 0x2, 0xde72d8, 0xc820262020, 0xd6b1c0, ...)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/deisctl/config/config.go:51 +0x46c
github.com/deis/deis/deisctl/config.Config(0x7fff5fbff54f, 0x4, 0x75bce8, 0x3, 0xc82043e540, 0x2, 0x2, 0xde72d8, 0xc820262020, 0x0, ...)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/deisctl/config/config.go:26 +0xcf
github.com/deis/deis/deisctl/cmd.Config(0x7fff5fbff54f, 0x4, 0x75bce8, 0x3, 0xc82043e540, 0x2, 0x2, 0xde72d8, 0xc820262020, 0x0, ...)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/deisctl/cmd/cmd.go:438 +0x96
github.com/deis/deis/deisctl/client.(*Client).Config(0xc8203508a0, 0xc820116800, 0x5, 0x8, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/deisctl/client/client.go:175 +0x3e8
main.Command(0xc820116800, 0x5, 0x8, 0xc820000180)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/deisctl/deisctl.go:115 +0xf4c
main.main()
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/deisctl/deisctl.go:22 +0x28

goroutine 33 [chan receive]:
golang.org/x/crypto/ssh.(*handshakeTransport).readPacket(0xc82014b680, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/handshake.go:124 +0x6d
golang.org/x/crypto/ssh.(*mux).onePacket(0xc820324540, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/mux.go:224 +0x5e
golang.org/x/crypto/ssh.(*mux).loop(0xc820324540)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/mux.go:199 +0x42
created by golang.org/x/crypto/ssh.newMux
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/mux.go:128 +0x20d

goroutine 38 [chan receive]:
golang.org/x/crypto/ssh/agent.ForwardToAgent.func1(0xc8202684e0, 0xda8370, 0xc82027e880)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/agent/forward.go:38 +0x54
created by golang.org/x/crypto/ssh/agent.ForwardToAgent
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/agent/forward.go:49 +0x147

goroutine 20 [IO wait]:
net.runtime_pollWait(0xda81c0, 0x72, 0xc820012280)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc82021fb80, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82021fb80, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82021fb20, 0xc820258000, 0x1000, 0x1000, 0x0, 0xd67050, 0xc820012280)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820112480, 0xc820258000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
bufio.(*Reader).fill(0xc82011d500)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Read(0xc82011d500, 0xc82026e1a0, 0x5, 0x5, 0x14, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:207 +0x260
io.ReadAtLeast(0xdac830, 0xc82011d500, 0xc82026e1a0, 0x5, 0x5, 0x5, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:298 +0xe6
io.ReadFull(0xdac830, 0xc82011d500, 0xc82026e1a0, 0x5, 0x5, 0xc820266610, 0x0, 0x0)
	/usr/local/go/src/io/io.go:316 +0x62
golang.org/x/crypto/ssh.(*streamPacketCipher).readPacket(0xc82026e180, 0xc800000009, 0xdac830, 0xc82011d500, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/cipher.go:137 +0xc1
golang.org/x/crypto/ssh.(*connectionState).readPacket(0xc8202326c0, 0xc82011d500, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/transport.go:110 +0xce
golang.org/x/crypto/ssh.(*transport).readPacket(0xc8202326c0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/transport.go:106 +0x55
golang.org/x/crypto/ssh.(*handshakeTransport).readOnePacket(0xc82014b680, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/handshake.go:153 +0xe7
golang.org/x/crypto/ssh.(*handshakeTransport).readLoop(0xc82014b680)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/handshake.go:133 +0x25
created by golang.org/x/crypto/ssh.newClientTransport
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/handshake.go:101 +0x31b

goroutine 34 [chan receive]:
golang.org/x/crypto/ssh.(*Client).handleGlobalRequests(0xc82026c340, 0xc820268420)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:137 +0x4a
created by golang.org/x/crypto/ssh.NewClient
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:54 +0xe4

goroutine 35 [chan receive]:
golang.org/x/crypto/ssh.(*Client).handleChannelOpens(0xc82026c340, 0xc820268360)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:146 +0x6e
created by golang.org/x/crypto/ssh.NewClient
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:55 +0x110

goroutine 36 [semacquire]:
sync.runtime_Syncsemacquire(0xc82026c310)
	/usr/local/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc82026c300)
	/usr/local/go/src/sync/cond.go:62 +0x9b
golang.org/x/crypto/ssh.(*mux).Wait(0xc820324540, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/mux.go:110 +0xb8
golang.org/x/crypto/ssh.NewClient.func1(0xc82026c340)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:57 +0x39
created by golang.org/x/crypto/ssh.NewClient
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:59 +0x132

goroutine 37 [chan receive]:
golang.org/x/crypto/ssh.(*forwardList).handleChannels(0xc82026c350, 0xc820268480)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/tcpip.go:178 +0x6e
created by golang.org/x/crypto/ssh.NewClient
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:60 +0x18f

goroutine 54 [chan receive]:
golang.org/x/crypto/ssh/agent.ForwardToAgent.func1(0xc820354480, 0xda8370, 0xc820350800)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/agent/forward.go:38 +0x54
created by golang.org/x/crypto/ssh/agent.ForwardToAgent
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/agent/forward.go:49 +0x147

goroutine 40 [IO wait]:
net.runtime_pollWait(0xda7f80, 0x72, 0xc820012280)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8203246f0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8203246f0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820324690, 0xc82026b000, 0x1000, 0x1000, 0x0, 0xd67050, 0xc820012280)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820262038, 0xc82026b000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
bufio.(*Reader).fill(0xc820268600)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Read(0xc820268600, 0xc82035a1a0, 0x5, 0x5, 0x14, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:207 +0x260
io.ReadAtLeast(0xdac830, 0xc820268600, 0xc82035a1a0, 0x5, 0x5, 0x5, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:298 +0xe6
io.ReadFull(0xdac830, 0xc820268600, 0xc82035a1a0, 0x5, 0x5, 0xc82034a500, 0x0, 0x0)
	/usr/local/go/src/io/io.go:316 +0x62
golang.org/x/crypto/ssh.(*streamPacketCipher).readPacket(0xc82035a180, 0xc800000009, 0xdac830, 0xc820268600, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/cipher.go:137 +0xc1
golang.org/x/crypto/ssh.(*connectionState).readPacket(0xc820326fc0, 0xc820268600, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/transport.go:110 +0xce
golang.org/x/crypto/ssh.(*transport).readPacket(0xc820326fc0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/transport.go:106 +0x55
golang.org/x/crypto/ssh.(*handshakeTransport).readOnePacket(0xc8202821e0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/handshake.go:153 +0xe7
golang.org/x/crypto/ssh.(*handshakeTransport).readLoop(0xc8202821e0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/handshake.go:133 +0x25
created by golang.org/x/crypto/ssh.newClientTransport
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/handshake.go:101 +0x31b

goroutine 49 [chan receive]:
golang.org/x/crypto/ssh.(*handshakeTransport).readPacket(0xc8202821e0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/handshake.go:124 +0x6d
golang.org/x/crypto/ssh.(*mux).onePacket(0xc8203fc540, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/mux.go:224 +0x5e
golang.org/x/crypto/ssh.(*mux).loop(0xc8203fc540)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/mux.go:199 +0x42
created by golang.org/x/crypto/ssh.newMux
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/mux.go:128 +0x20d

goroutine 50 [chan receive]:
golang.org/x/crypto/ssh.(*Client).handleGlobalRequests(0xc8203f62c0, 0xc8203543c0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:137 +0x4a
created by golang.org/x/crypto/ssh.NewClient
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:54 +0xe4

goroutine 51 [chan receive]:
golang.org/x/crypto/ssh.(*Client).handleChannelOpens(0xc8203f62c0, 0xc820354300)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:146 +0x6e
created by golang.org/x/crypto/ssh.NewClient
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:55 +0x110

goroutine 52 [semacquire]:
sync.runtime_Syncsemacquire(0xc8203f6290)
	/usr/local/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc8203f6280)
	/usr/local/go/src/sync/cond.go:62 +0x9b
golang.org/x/crypto/ssh.(*mux).Wait(0xc8203fc540, 0x0, 0x0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/mux.go:110 +0xb8
golang.org/x/crypto/ssh.NewClient.func1(0xc8203f62c0)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:57 +0x39
created by golang.org/x/crypto/ssh.NewClient
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:59 +0x132

goroutine 53 [chan receive]:
golang.org/x/crypto/ssh.(*forwardList).handleChannels(0xc8203f62d0, 0xc820354420)
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/tcpip.go:178 +0x6e
created by golang.org/x/crypto/ssh.NewClient
	/Users/jenkins/workspace/build-deisctl-installer-darwin/src/github.com/deis/deis/Godeps/_workspace/src/golang.org/x/crypto/ssh/client.go:60 +0x18f
```

--- a/deisctl/config/config.go
+++ b/deisctl/config/config.go
@@ -6,7 +6,7 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
-	"strings"
+	"regexp"
 
 	"github.com/deis/deis/deisctl/utils"
 )
@@ -65,12 +65,17 @@ func doConfig(target string, action string, key []string, cb Backend, w io.Write
 
 func doConfigSet(cb Backend, root string, kvs []string) ([]string, error) {
 	var result []string
+	regex := regexp.MustCompile(`^(.+)=([\s\S]+)$`)
 
 	for _, kv := range kvs {
 
+		if !regex.MatchString(kv) {
+			return []string{}, fmt.Errorf("'%s' does not match the pattern 'key=var', ex: foo=bar\n", kv)
+		}
+
 		// split k/v from args
-		split := strings.SplitN(kv, "=", 2)
-		k, v := split[0], split[1]
+		captures := regex.FindStringSubmatch(kv)
+		k, v := captures[1], captures[2]
 
 		// prepare path and value
 		path := root + k

--- a/deisctl/config/config_test.go
+++ b/deisctl/config/config_test.go
@@ -3,7 +3,6 @@ package config
 import (
 	"bytes"
 	"encoding/base64"
-	"fmt"
 	"io/ioutil"
 	"os"
 	"testing"
@@ -27,7 +26,7 @@ func TestGetConfig(t *testing.T) {
 	expected := "foo\n8000\n"
 	output := testWriter.String()
 	if output != expected {
-		t.Error(fmt.Errorf("Expected: '%s', Got:'%s'", expected, output))
+		t.Errorf("Expected: '%s', Got:'%s'", expected, output)
 	}
 }
 
@@ -59,7 +58,21 @@ func TestSetConfig(t *testing.T) {
 	expected := "bar\n1000\n"
 	output := testWriter.String()
 	if output != expected {
-		t.Error(fmt.Errorf("Expected: '%s', Got:'%s'", expected, output))
+		t.Errorf("Expected: '%s', Got:'%s'", expected, output)
+	}
+}
+
+func TestSetConfigError(t *testing.T) {
+	t.Parallel()
+
+	testMock := mock.ConfigBackend{Expected: []*model.ConfigNode{}}
+	testWriter := bytes.Buffer{}
+
+	expected := "'foo' does not match the pattern 'key=var', ex: foo=bar\n"
+	err := doConfig("controller", "set", []string{"foo", "=", "bar"}, testMock, &testWriter)
+
+	if err.Error() != expected {
+		t.Errorf("Expected: '%s', Got:'%q'", expected, err)
 	}
 }
 
@@ -78,7 +91,7 @@ func TestDeleteConfig(t *testing.T) {
 	expected := "testing\nport\n"
 	output := testWriter.String()
 	if output != expected {
-		t.Error(fmt.Errorf("Expected: '%s', Got:'%s'", expected, output))
+		t.Errorf("Expected: '%s', Got:'%s'", expected, output)
 	}
 }
 

